app/models/concerns/course/search_concern.rb

```diff
@@ -6,7 +6,7 @@ module ClassMethods
     # @param [String] keywords The keywords for filtering courses.
     # @return [Array<Course>] The courses which match the keyword. All courses will be returned if
     #   keyword is blank.
-    def search(keyword)
+    def search(keyword) # rubocop:disable Metrics/AbcSize
```

Comment:
It's a complete scope with sql joins, I think keep this method un-splitted makes more sense ?